### ISSUE TITLE
'CompareSymbolsCorrectlyFix' encountered an error and has been disabled

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn-analyzers#4116]
### Analyzer

**Diagnostic ID**: [RS1024]: `Compare symbols correctly`

### Analyzer source

**NuGet Package**: https://www.nuget.org/packages/Microsoft.CodeAnalysis.Analyzers/3.3.0

**Version**: v3.3.0 (Latest)

### Describe the bug

'CompareSymbolsCorrectlyFix' encountered an error and has been disabled

### Steps To Reproduce

Place the caret on the warning highlighting on the `.Equals(t2)` call and try to use the `Compare symbols correctly` fix
```
  class C
  {
    void M()
    {
      INamedTypeSymbol? ts1 = null;
      INamedTypeSymbol? ts2 = null;
      var result = ts1.Equals(ts2);
    }
  }
```

### Expected behavior
No exceptions

### Actual behavior
```
System.InvalidCastException : Unable to cast object of type 'Microsoft.CodeAnalysis.CSharp.Syntax.MemberAccessExpressionSyntax' to type 'Microsoft.CodeAnalysis.CSharp.Syntax.ParameterSyntax'.
   at System.Linq.Enumerable.<CastIterator>d__97`1.MoveNext()
   at Microsoft.CodeAnalysis.CSharp.SyntaxFactory.SeparatedList[TNode](IEnumerable`1 nodes)
   at Microsoft.CodeAnalysis.CSharp.CodeGeneration.CSharpSyntaxGenerator.AsParameterList(IEnumerable`1 parameters)
   at Microsoft.CodeAnalysis.CSharp.CodeGeneration.CSharpSyntaxGenerator.InsertParameters(SyntaxNode declaration,Int32 index,IEnumerable`1 parameters)
   at Microsoft.CodeAnalysis.Editing.SyntaxGenerator.AddParameters(SyntaxNode declaration,IEnumerable`1 parameters)
   at async Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.Fixers.CompareSymbolsCorrectlyFix.EnsureEqualsCorrectAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.Fixers.CompareSymbolsCorrectlyFix.ConvertToEqualsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.GetChangedSolutionAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.ComputeOperationsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.GetPreviewOperationsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedAction.GetPreviewResultAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedActionWithNestedFlavors.<>c__DisplayClass11_0.<GetPreviewAsync>b__0(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Extensions.IExtensionManagerExtensions.PerformFunctionAsync[T](<Unknown Parameters>)
   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
```

**Comment by Evangelink** [comment]
Hi @TessenR,

Thanks for the reproducer and the bug report. I confirm that I was able to reproduce the issue on v3.3.0 and also that this issue is already fixed in the latest (non released) version of the code. I have done a unit test and also an actual test using the latest beta (3.3.1-beta1.20454.1).

Unit test for reference:

```csharp
[Fact]
        [WorkItem(4116, "https://github.com/dotnet/roslyn-analyzers/issues/4116")]
        public async Task InvalidCast_NotReproduced()
        {
            var source = @"
#nullable enable
using Microsoft.CodeAnalysis;

class C
{
    void M()
    {
        INamedTypeSymbol? ts1 = null;
        INamedTypeSymbol? ts2 = null;
        var result = [|{|CS8602:ts1|}.Equals(ts2)|];
    }
}";

            var fixedSource = @"
#nullable enable
using Microsoft.CodeAnalysis;

class C
{
    void M()
    {
        INamedTypeSymbol? ts1 = null;
        INamedTypeSymbol? ts2 = null;
        var result = SymbolEqualityComparer.Default.Equals(ts1, ts2);
    }
}";

            await new VerifyCS.Test
            {
                LanguageVersion = CSharp.LanguageVersion.CSharp8,
                TestState = { Sources = { source, SymbolEqualityComparerStubCSharp } },
                FixedState = { Sources = { fixedSource, SymbolEqualityComparerStubCSharp } },
            }.RunAsync();
        }
```

Tagging @mavasani for an ETA for the next release.
